cmd/libsnap-confine-private/mountinfo.c

```diff
@@ -156,67 +156,81 @@ static char *parse_next_string_field(sc_mountinfo_entry * entry,
 	size_t input_idx = 0;
 	size_t output_idx = 0;
 
-	// Scan characters until we run out of memory to scan or we find a space.
-	// The kernel uses simple octal escape sequences for the following:
-	// space, tab, newline, backwards slash. Everything else is copied
-	// verbatim.
-	bool stop = false;
-	for (int c; c = input[input_idx], c != '\0' && !stop;
-	     input_idx++, output_idx++) {
-		switch (c) {
-		case ' ':
-			// Fields are space delimited. Represent the space
-			// as the end-of-string marker, skip over it, and stop parsing.
-			output[output_idx] = '\0';
-			stop = true;
+	// Scan characters until we run out of memory to scan or we find a
+	// space.  The kernel uses simple octal escape sequences for the
+	// following: space, tab, newline, backwards slash. Everything else is
+	// copied verbatim.
+	for (;;) {
+		int c = input[input_idx];
+		if (c == '\0') {
+			// The string is over before we see anything then
+			// return NULL. This is an indication of end-of-input
+			// to the caller.
+			if (output_idx == 0) {
+			    return NULL;
+			}
+			// The scanned line is NUL terminated. This ensures that the
+			// terminator is copied to the output buffer.
+			output[output_idx++] = '\0';
+			// NOTE: we are no longer advancing the reading index
+			// because we reached the end of the buffer.
+			break;
+		} else if (c == ' ') {
+			// Fields are space delimited or end-of-string terminated.
+			// Represent either as the end-of-string marker, skip over it,
+			// and stop parsing.
+			output[output_idx++] = '\0';
+			// Advance the reading index by one.
```

Comment:
Nitpick: this comment on it's own is obvious. Did you mean to say something more about why you are advancing it?

Maybe better is:
```
// Fields are space delimited or end-of-string terminated.
// Represent either as the end-of-string marker, skip over it,
// and stop parsing by terminating the output , then breaking
// out of the loop but advancing the reading index which is
// needed for subsequent calls.
output[output_idx++] = '\0';
input_idx++;
break;
```

Not a blocker.

Reply:
I agree this was a `// this is a comment` comment.

Reply:
I merged the comments around this section as suggested now.